### ISSUE TITLE
bundle outdated prints incorrect output

### ISSUE THREAD

**Issue by jaredbeck** [bundler/bundler#5201]
## Overview

`bundle outdated` fails to list gems that are outdated.

## Expected output

```
bundle outdated
Fetching gem metadata from https://rubygems.org/............
Fetching version metadata from https://rubygems.org/.
Resolving dependencies...

Outdated gems included in the bundle:
  * aws-sdk (newest 2.6.27, installed 2.6.23) in group "default"
  * aws-sdk-core (newest 2.6.27, installed 2.6.23)
  * aws-sdk-resources (newest 2.6.27, installed 2.6.23)
```

## Actual output

```
bundle outdated
Fetching gem metadata from https://rubygems.org/............
Fetching version metadata from https://rubygems.org/.
Resolving dependencies...

Bundle up to date!
```

## Diff

```diff
diff expected actual
6,9c6
< Outdated gems included in the bundle:
<   * aws-sdk (newest 2.6.27, installed 2.6.23) in group "default"
<   * aws-sdk-core (newest 2.6.27, installed 2.6.23)
<   * aws-sdk-resources (newest 2.6.27, installed 2.6.23)
---
> Bundle up to date!
```

## Full steps to reliably reproduce the problem:

1. Install gems until you get *exactly* the following output from `gem list aws`. 

```
gem list aws

*** LOCAL GEMS ***

aws-sdk (2.6.23)
aws-sdk-core (2.6.23)
aws-sdk-resources (2.6.23)
```

Note that aws-sdk 2.6.27 is *not* installed.

2. In an empty directory, create the following files:

```
cat Gemfile Gemfile.lock 
# Gemfile
source 'https://rubygems.org'
ruby '2.2.5'
gem 'aws-sdk'

# Gemfile.lock
GEM
  remote: https://rubygems.org/
  specs:
    aws-sdk (2.6.23)
      aws-sdk-resources (= 2.6.23)
    aws-sdk-core (2.6.23)
      aws-sigv4 (~> 1.0)
      jmespath (~> 1.0)
    aws-sdk-resources (2.6.23)
      aws-sdk-core (= 2.6.23)
    aws-sigv4 (1.0.0)
    jmespath (1.3.1)

PLATFORMS
  ruby

DEPENDENCIES
  aws-sdk

RUBY VERSION
   ruby 2.2.5p319

BUNDLED WITH
   1.13.6
```

Notice that the `Gemfile` places no constraints on `aws-sdk`.

3. Run `bundle outdated` and see the message "Bundle up to date!", which is incorrect, because aws-sdk 2.6.27 has been released. 

## Versions Tested

```bash
ruby -v # ruby 2.2.5p319 (2016-04-26 revision 54774) [x86_64-darwin15]
gem -v # 2.4.5.1
bundler -v # Bundler version 1.13.6
rbenv -v # rbenv 1.0.0-21-g9fdce5d
```

## Notes

- `bundle outdated --no-strict` also erroneously prints "Bundle up to date!"
- `bundle update` works as expected, updating the gems to 2.6.27 and writing a new lock file
- If you `gem install aws-sdk -v '2.6.27'` and then run `bundle outdated`, it produces the correct output.
- The aws gems are just the example I have chosen. This issue is not specific to them

**Comment by chrismo** [comment]
This looks like #4979. If you agree, we can close this.

**Comment by jaredbeck** [comment]
> This looks like #4979. If you agree, we can close this.

I'm not certain it's the same. I'm not using `source .. do` as you are in your [test case](https://github.com/chrismo/bundler-cases/commit/4c3ea5d3eef1b7abfd80c6626c5bbc38d4b591f5) but I don't know if that's a significant difference.

So this is fixed in master, but has not been released yet? If you're convinced it's fixed, feel free to close.

**Comment by chrismo** [comment]
Correct, fixed in master, should go out with 1.14.

**Comment by jaredbeck** [comment]
As a workaround, instead of `bundle outdated`, I have been using:

```bash
bundle update && git checkout . && bundle outdated
```

Hope it helps someone.
